File: android-app/data/src/main/java/com/starlight/module/data/datasource/mappers/UserMapper.kt
package com.starlight.module.data.datasource.mappers

import com.starlight.module.data.datasource.entity.UserEntity
import com.starlight.module.domain.mappers.DomainMapper
import com.starlight.module.domain.model.User
import com.starlight.module.domain.utils.Constant
import java.util.*

class UserMapper : DomainMapper<UserEntity, User> {
    override fun toDomain(entity: UserEntity): User {
        return User(
            entity.userName,
            entity.userId,
            entity.firstName,
            entity.lastName,
            entity.email,
            entity.phoneNumber,
            entity.isActive,
            entity.avatar ?: "",
            entity.gender,
            Date(),
            User.NORMAL_USER
        )
    }

    override fun fromDomain(model: User): UserEntity {
        return UserEntity(
            model.userName,
            model.userId,
            model.firstName,
            model.lastName,
            model.email,
            model.phoneNumber,
            model.isActive,
            model.avatar,
            model.gender,
            Constant.simpleDateFormat.format(model.dob),
            User.NORMAL_USER
        )
    }
}

File: nest-cloud/src/data/mappers/UserMapper.ts
import DomainMapper from "../../domain/mappers/DomainMapper";
import User from "../../domain/model/User";
import { UserEntity } from "../entity/UserEntity";

export class UserMapper implements DomainMapper<User, UserEntity>{
    toDomain(entity: UserEntity): User {
        return new User(
            entity.user_id,
            entity.first_name,
            entity.last_name,
            entity.user_name,
            entity.email,
            entity.phone_number,
            entity.avatar,
            entity.gender,
            entity.is_active,
            entity.dob.toString(),
            entity.permission_id
        );
    }
    fromDomain(domain: User): UserEntity {
        return new UserEntity(
            domain.user_id,
            domain.first_name,
            domain.last_name,
            domain.user_name,
            domain.email,
            domain.phone_number,
            "",
            "",
            "",
            domain.avatar,
            domain.gender,
            domain.is_active,
            domain.dob,
            domain.permission_id
        )
    }
}